docs/sphinx/user_guide/feature/reduction.rst

```diff
@@ -159,3 +159,108 @@ Reduction Policies
 For more information about available RAJA reduction policies and guidance
 on which to use with RAJA execution policies, please see 
 :ref:`reducepolicy-label`.
+
+--------------------------------
+Experimental Reduction Interface
+--------------------------------
+
+An experimental reduction interface is now available that hopes to improve
+upon the current reduction model in RAJA. This interface allows ``RAJA::forall``
+to take optional "plugin-like" objects to extend the execution behaviour 
+of a ``RAJA::forall`` execution context.
+
+
+RAJA::expt::Reduce
+..................
+::
+
+  double rs;
+  double rm;
+      
+  RAJA::forall<EXEC_POL> ( Res, Seg, 
+  RAJA::expt::Reduce<RAJA::operators::plus>(&rs),
+  RAJA::expt::Reduce<RAJA::operators::minimum>(&rm),
+  [=] (int i, double& _rs, double& _rm) {
+    _rs += ...
+    _rm = RAJA_MIN(..., _rm); 
+  }
+  );
+  
+  std::cout << rs ...
+  std::cout << rm ...
+
+* ``RAJA::expt::Reduce`` takes a target variable to write the final result to 
```

Comment:
I think we should be clear that Reduce combines its value with the existing value instead of just overwriting it.
```suggestion
* ``RAJA::expt::Reduce`` takes a target variable to add the final result to 
```

Reply:
Hmm, it might be worth having that be an option, extra arg to reduce for initial value that defaults to the original?

Reply:
Done.